boards: arc: nsim_em7d_v22: switch to ns16550 UART model

IRISZZW opened the pull request:
In PR #26836, we switch nSIM from custom legacy ARC UART model
to ns16550 model, which will allow us to use zephyr images build for
nSIM on other platforms like HAPS, QEMU, etc...
In PR #27334, which introduce new board nsim_em7d_v22, has gone
parallel to the switch to dwuart, and is still using legacy model.
With wrong configuration, the uart for nsim_em7d_v22 has no output,
which cause all tests failure.

Fixes: #28976

IRISZZW commented:
sanity check results with this fix, 1 failure is timeout failure, and I run it manually，it passed.

```
./scripts/sanitycheck -p nsim_em7d_v22 -O build.sanity.nsim_em7d_v22 -n -v
Keeping artifacts untouched
INFO    - JOBS: 4
INFO    - Building initial testcase list...
INFO    - 928 test configurations selected, 696 configurations discarded due to filters.
INFO    - Adding tasks to the queue...
INFO    -   1/232 nsim_em7d_v22             samples/smp/pi/sample.smp.pi                       SKIPPED (filter)
INFO    -   2/232 nsim_em7d_v22             samples/subsys/shell/shell_module/sample.shell.shell_module.minimal_rtt SKIPPED (filter)
INFO    -   3/232 nsim_em7d_v22             samples/scheduler/metairq_dispatch/sample.kernel.metairq_dispatch PASSED (nsim 3.017s)
INFO    -   4/232 nsim_em7d_v22             samples/userspace/shared_mem/sample.kernel.memory_protection.shared_mem PASSED (nsim 2.440s)
INFO    -   5/232 nsim_em7d_v22             samples/userspace/prod_consumer/sample.userspace.prod_consumer PASSED (nsim 3.377s)
INFO    -   6/232 nsim_em7d_v22             samples/posix/eventfd/sample.posix.eventfd         PASSED (nsim 2.287s)
INFO    -   7/232 nsim_em7d_v22             samples/subsys/tracing/tracing.osawareness.openocd PASSED (nsim 2.870s)
INFO    -   8/232 nsim_em7d_v22             samples/subsys/mgmt/osdp/sample.mgmt.osdp          SKIPPED (filter)
INFO    -   9/232 nsim_em7d_v22             samples/subsys/shell/shell_module/sample.shell.shell_module.minimal PASSED (build)
INFO    -  10/232 nsim_em7d_v22             samples/subsys/logging/logger/sample.logger.rtt    SKIPPED (filter)
INFO    -  11/232 nsim_em7d_v22             samples/synchronization/sample.kernel.synchronization PASSED (nsim 2.457s)
INFO    -  12/232 nsim_em7d_v22             samples/subsys/shell/shell_module/sample.shell.shell_module PASSED (build)
INFO    -  13/232 nsim_em7d_v22             samples/subsys/console/getline/sample.console.getline PASSED (build)
INFO    -  14/232 nsim_em7d_v22             samples/subsys/console/getchar/sample.console.getchar PASSED (build)
INFO    -  15/232 nsim_em7d_v22             samples/subsys/logging/logger/sample.logger.usermode PASSED (build)
INFO    -  16/232 nsim_em7d_v22             samples/subsys/logging/logger/sample.logger.basic  PASSED (nsim 2.926s)
INFO    -  17/232 nsim_em7d_v22             samples/subsys/console/echo/sample.console.echo    PASSED (build)
INFO    -  18/232 nsim_em7d_v22             samples/sensor/sensor_shell/sample.sensor.shell    PASSED (build)
INFO    -  19/232 nsim_em7d_v22             samples/sensor/thermometer/sample.sensor.thermometer PASSED (build)
INFO    -  20/232 nsim_em7d_v22             samples/portability/cmsis_rtos_v2/timer_synchronization/sample.portability.cmsis_rtos_v2.timer_synchronization PASSED (nsim 3.820s)
INFO    -  21/232 nsim_em7d_v22             samples/portability/cmsis_rtos_v2/philosophers/sample.portability.cmsis_rtos_v2.philosopher.same_prio PASSED (nsim 3.198s)
INFO    -  22/232 nsim_em7d_v22             samples/portability/cmsis_rtos_v2/philosophers/sample.portability.cmsis_rtos_v2.philosopher.semaphores PASSED (nsim 3.856s)
INFO    -  23/232 nsim_em7d_v22             samples/portability/cmsis_rtos_v2/philosophers/sample.portability.cmsis_rtos_v2.philosopher PASSED (nsim 3.206s)
INFO    -  24/232 nsim_em7d_v22             samples/portability/cmsis_rtos_v1/timer_synchronization/sample.portability.cmsis_rtos_v1.timer_synchronization PASSED (nsim 2.646s)
INFO    -  25/232 nsim_em7d_v22             samples/portability/cmsis_rtos_v1/philosophers/sample.portability.cmsis_rtos_v1.philosopher.same_prio PASSED (nsim 3.059s)
INFO    -  26/232 nsim_em7d_v22             samples/portability/cmsis_rtos_v1/philosophers/sample.portability.cmsis_rtos_v1.philosopher.semaphores PASSED (nsim 3.797s)
INFO    -  27/232 nsim_em7d_v22             samples/philosophers/sample.kernel.philosopher.coop_only PASSED (nsim 8.651s)
INFO    -  28/232 nsim_em7d_v22             samples/portability/cmsis_rtos_v1/philosophers/sample.portability.cmsis_rtos_v1.philosopher PASSED (nsim 3.797s)
INFO    -  29/232 nsim_em7d_v22             samples/philosophers/sample.kernel.philosopher.preempt_only PASSED (nsim 3.433s)
INFO    -  30/232 nsim_em7d_v22             samples/philosophers/sample.kernel.philosopher.fifos PASSED (nsim 3.429s)
INFO    -  31/232 nsim_em7d_v22             samples/philosophers/sample.kernel.philosopher.lifos PASSED (nsim 3.580s)
INFO    -  32/232 nsim_em7d_v22             samples/philosophers/sample.kernel.philosopher.stacks PASSED (nsim 4.040s)
INFO    -  33/232 nsim_em7d_v22             samples/drivers/flash_shell/sample.drivers.flash.shell SKIPPED (filter)
INFO    -  34/232 nsim_em7d_v22             samples/drivers/entropy/sample.drivers.entropy     SKIPPED (filter)
INFO    -  35/232 nsim_em7d_v22             samples/philosophers/sample.kernel.philosopher.semaphores PASSED (nsim 3.303s)
INFO    -  36/232 nsim_em7d_v22             samples/philosophers/sample.kernel.philosopher.static PASSED (nsim 3.133s)
INFO    -  37/232 nsim_em7d_v22             samples/philosophers/sample.kernel.philosopher.same_prio PASSED (nsim 3.132s)
INFO    -  38/232 nsim_em7d_v22             samples/philosophers/sample.kernel.philosopher     PASSED (nsim 2.418s)
INFO    -  39/232 nsim_em7d_v22             samples/hello_world/sample.basic.helloworld        PASSED (nsim 3.510s)
INFO    -  40/232 nsim_em7d_v22             samples/drivers/crypto/sample.drivers.crypto.mbedtls.micro PASSED (nsim 3.772s)
INFO    -  41/232 nsim_em7d_v22             samples/cpp_synchronization/sample.cpp.synchronization PASSED (nsim 2.924s)
INFO    -  42/232 nsim_em7d_v22             samples/application_development/out_of_tree_driver/sample.drivers.out_of_tree PASSED (nsim 3.872s)
INFO    -  43/232 nsim_em7d_v22             samples/drivers/crypto/sample.drivers.crypto.mbedtls PASSED (nsim 3.409s)
INFO    -  44/232 nsim_em7d_v22             samples/application_development/external_lib/sample.app_dev.external_lib PASSED (nsim 3.693s)
INFO    -  45/232 nsim_em7d_v22             tests/portability/cmsis_rtos_v2/portability.cmsis_rtos_v2 PASSED (nsim 3.318s)
INFO    -  46/232 nsim_em7d_v22             tests/deprecated/inttype/deprecated.inttypes       PASSED (build)
INFO    -  47/232 nsim_em7d_v22             tests/portability/cmsis_rtos_v1/portability.cmsis_rtos_v1 PASSED (nsim 3.779s)
INFO    -  48/232 nsim_em7d_v22             tests/subsys/debug/coredump/coredump.logging_backend SKIPPED (filter)
INFO    -  49/232 nsim_em7d_v22             tests/ztest/custom_output/testing.ztest.customized_output PASSED (nsim 2.854s)
INFO    -  50/232 nsim_em7d_v22             tests/ztest/custom_output/testing.ztest.regular_output PASSED (nsim 2.728s)
INFO    -  51/232 nsim_em7d_v22             tests/ztest/mock/testing.ztest.mock                PASSED (nsim 2.202s)
INFO    -  52/232 nsim_em7d_v22             tests/ztest/base/testing.ztest.verbose_2           PASSED (nsim 2.347s)
INFO    -  53/232 nsim_em7d_v22             tests/ztest/base/testing.ztest.verbose_1           PASSED (nsim 5.649s)
INFO    -  54/232 nsim_em7d_v22             tests/ztest/base/testing.ztest.verbose_0           PASSED (nsim 2.894s)
INFO    -  55/232 nsim_em7d_v22             tests/subsys/shell/shell_history/shell.history     PASSED (nsim 2.321s)
INFO    -  56/232 nsim_em7d_v22             tests/subsys/logging/log_core_additional/logging.add.sync PASSED (nsim 2.425s)
INFO    -  57/232 nsim_em7d_v22             tests/subsys/logging/log_core_additional/logging.add.async PASSED (nsim 6.339s)
INFO    -  58/232 nsim_em7d_v22             tests/subsys/logging/log_immediate/logging.log_immediate.clean_output PASSED (nsim 5.034s)
INFO    -  59/232 nsim_em7d_v22             tests/subsys/logging/log_output/logging.log_output PASSED (nsim 2.279s)
INFO    -  60/232 nsim_em7d_v22             tests/subsys/logging/log_immediate/logging.log_immediate PASSED (nsim 4.933s)
INFO    -  61/232 nsim_em7d_v22             tests/subsys/logging/log_msg/logging.log_msg       PASSED (nsim 2.221s)
INFO    -  62/232 nsim_em7d_v22             tests/subsys/logging/log_list/logging.log_list     PASSED (nsim 2.383s)
INFO    -  63/232 nsim_em7d_v22             tests/subsys/logging/log_core/logging.log_core     PASSED (nsim 2.378s)
INFO    -  64/232 nsim_em7d_v22             tests/subsys/jwt/libraries.encoding.jwt            PASSED (nsim 15.326s)
INFO    -  65/232 nsim_em7d_v22             tests/subsys/fs/fs_api/filesystem.api              PASSED (nsim 2.593s)
INFO    -  66/232 nsim_em7d_v22             tests/shell/shell.core                             PASSED (nsim 3.195s)
INFO    -  67/232 nsim_em7d_v22             tests/posix/eventfd_basic/posix.eventfd_basic.posix_api PASSED (nsim 2.479s)
INFO    -  68/232 nsim_em7d_v22             tests/posix/eventfd_basic/posix.eventfd_basic      PASSED (nsim 4.165s)
INFO    -  69/232 nsim_em7d_v22             tests/posix/eventfd/portability.posix.eventfd      PASSED (nsim 2.547s)
INFO    -  70/232 nsim_em7d_v22             tests/misc/test_build/buildsystem.kconfig.utf8_in_values PASSED (build)
INFO    -  71/232 nsim_em7d_v22             tests/posix/fs/portability.posix.fs.newlib         PASSED (nsim 2.703s)
INFO    -  72/232 nsim_em7d_v22             tests/posix/fs/portability.posix.fs                PASSED (nsim 2.691s)
INFO    -  73/232 nsim_em7d_v22             tests/misc/test_build/buildsystem.debug.build      PASSED (build)
INFO    -  74/232 nsim_em7d_v22             tests/posix/common/portability.posix.common.newlib PASSED (nsim 8.696s)
INFO    -  75/232 nsim_em7d_v22             tests/lib/heap/lib.heap                            SKIPPED (filter)
INFO    -  76/232 nsim_em7d_v22             tests/posix/common/portability.posix.common        PASSED (nsim 9.100s)
INFO    -  77/232 nsim_em7d_v22             tests/lib/onoff/libraries.onoff                    PASSED (nsim 2.392s)
INFO    -  78/232 nsim_em7d_v22             tests/lib/cmsis_dsp/transform/libraries.cmsis_dsp.transform.rf32 SKIPPED (filter)
INFO    -  79/232 nsim_em7d_v22             tests/lib/notify/libraries.data_structures         PASSED (nsim 2.357s)
INFO    -  80/232 nsim_em7d_v22             tests/lib/cmsis_dsp/transform/libraries.cmsis_dsp.transform.rq15 SKIPPED (filter)
INFO    -  81/232 nsim_em7d_v22             tests/lib/cmsis_dsp/transform/libraries.cmsis_dsp.transform.cq15 SKIPPED (filter)
INFO    -  82/232 nsim_em7d_v22             tests/lib/heap_align/lib.heap_align                PASSED (nsim 9.241s)
INFO    -  83/232 nsim_em7d_v22             tests/lib/cmsis_dsp/svm/libraries.cmsis_dsp.svm    SKIPPED (filter)
INFO    -  84/232 nsim_em7d_v22             tests/lib/cmsis_dsp/support/libraries.cmsis_dsp.support SKIPPED (filter)
INFO    -  85/232 nsim_em7d_v22             tests/lib/fdtable/libraries.os.fdtable             PASSED (nsim 2.746s)
INFO    -  86/232 nsim_em7d_v22             tests/lib/cmsis_dsp/statistics/libraries.cmsis_dsp.statistics SKIPPED (filter)
INFO    -  87/232 nsim_em7d_v22             tests/lib/devicetree/legacy_api/libraries.devicetree.legacy PASSED (nsim 3.677s)
INFO    -  88/232 nsim_em7d_v22             tests/lib/devicetree/api/libraries.devicetree      PASSED (nsim 5.160s)
INFO    -  89/232 nsim_em7d_v22             tests/lib/cmsis_dsp/matrix/libraries.cmsis_dsp.matrix.binary_f32 SKIPPED (filter)
INFO    -  90/232 nsim_em7d_v22             tests/lib/cmsis_dsp/matrix/libraries.cmsis_dsp.matrix.binary_q31 SKIPPED (filter)
INFO    -  91/232 nsim_em7d_v22             tests/lib/cmsis_dsp/matrix/libraries.cmsis_dsp.matrix.binary_q15 SKIPPED (filter)
INFO    -  92/232 nsim_em7d_v22             tests/lib/cmsis_dsp/matrix/libraries.cmsis_dsp.matrix.unary_f64 SKIPPED (filter)
INFO    -  93/232 nsim_em7d_v22             tests/lib/cmsis_dsp/matrix/libraries.cmsis_dsp.matrix.unary_q31 SKIPPED (filter)
INFO    -  94/232 nsim_em7d_v22             tests/lib/cmsis_dsp/matrix/libraries.cmsis_dsp.matrix.unary_f32 SKIPPED (filter)
INFO    -  95/232 nsim_em7d_v22             tests/lib/cmsis_dsp/matrix/libraries.cmsis_dsp.matrix.unary_q15 SKIPPED (filter)
INFO    -  96/232 nsim_em7d_v22             tests/lib/cmsis_dsp/filtering/libraries.cmsis_dsp.filtering.misc SKIPPED (filter)
INFO    -  97/232 nsim_em7d_v22             tests/lib/cmsis_dsp/filtering/libraries.cmsis_dsp.filtering.biquad SKIPPED (filter)
INFO    -  98/232 nsim_em7d_v22             tests/lib/cmsis_dsp/filtering/libraries.cmsis_dsp.filtering.fir SKIPPED (filter)
INFO    -  99/232 nsim_em7d_v22             tests/lib/cmsis_dsp/fastmath/libraries.cmsis_dsp.fastmath SKIPPED (filter)
INFO    - 100/232 nsim_em7d_v22             tests/lib/cmsis_dsp/distance/libraries.cmsis_dsp.distance SKIPPED (filter)
INFO    - 101/232 nsim_em7d_v22             tests/lib/cmsis_dsp/complexmath/libraries.cmsis_dsp.complexmath SKIPPED (filter)
INFO    - 102/232 nsim_em7d_v22             tests/lib/cmsis_dsp/bayes/libraries.cmsis_dsp.bayes SKIPPED (filter)
INFO    - 103/232 nsim_em7d_v22             tests/lib/cmsis_dsp/basicmath/libraries.cmsis_dsp.basicmath SKIPPED (filter)
INFO    - 104/232 nsim_em7d_v22             tests/kernel/fatal/exception/kernel.common.stack_protection_arm_fpu_sharing SKIPPED (filter)
INFO    - 105/232 nsim_em7d_v22             tests/lib/sprintf/libraries.libc.sprintf           PASSED (nsim 3.388s)
INFO    - 106/232 nsim_em7d_v22             tests/lib/ringbuffer/libraries.data_structures     PASSED (nsim 3.579s)
INFO    - 107/232 nsim_em7d_v22             tests/lib/json/libraries.encoding.json             PASSED (nsim 3.559s)
INFO    - 108/232 nsim_em7d_v22             tests/lib/mem_alloc/libraries.libc.newlib          PASSED (nsim 3.893s)
INFO    - 109/232 nsim_em7d_v22             tests/kernel/fatal/message_capture/kernel.logging.message_capture PASSED (nsim 5.229s)
INFO    - 110/232 nsim_em7d_v22             tests/lib/c_lib/libraries.libc                     PASSED (nsim 3.815s)
INFO    - 111/232 nsim_em7d_v22             tests/kernel/fpu_sharing/generic/kernel.fpu_sharing.generic.arm SKIPPED (filter)
INFO    - 112/232 nsim_em7d_v22             tests/kernel/xip/arch.common.xip                   SKIPPED (filter)
INFO    - 113/232 nsim_em7d_v22             tests/kernel/workq/critical/kernel.workqueue.sam   SKIPPED (filter)
INFO    - 114/232 nsim_em7d_v22             tests/lib/mem_alloc/libraries.libc.minimal.mem_alloc PASSED (nsim 4.178s)
INFO    - 115/232 nsim_em7d_v22             tests/kernel/fatal/exception/kernel.common.stack_sentinel PASSED (nsim 4.179s)
INFO    - 116/232 nsim_em7d_v22             tests/kernel/fpu_sharing/generic/kernel.fpu_sharing.generic.arc PASSED (build)
INFO    - 117/232 nsim_em7d_v22             tests/kernel/workq/critical/kernel.workqueue       PASSED (nsim 5.051s)
INFO    - 118/232 nsim_em7d_v22             tests/kernel/fatal/exception/kernel.common.stack_protection PASSED (nsim 7.132s)
INFO    - 119/232 nsim_em7d_v22             tests/kernel/timer/starve/kernel.timer.starve      PASSED (build)
INFO    - 120/232 nsim_em7d_v22             tests/kernel/stack/stack/kernel.stack.usage        PASSED (nsim 6.886s)
INFO    - 121/232 nsim_em7d_v22             tests/kernel/workq/work_queue/kernel.workqueue     PASSED (nsim 6.218s)
INFO    - 122/232 nsim_em7d_v22             tests/kernel/tickless/tickless/kernel.tickless     SKIPPED (filter)
INFO    - 123/232 nsim_em7d_v22             tests/kernel/timer/timer_monotonic/kernel.timer.monotonic PASSED (nsim 17.553s)
INFO    - 124/232 nsim_em7d_v22             tests/kernel/workq/work_queue_api/kernel.workqueue.api PASSED (nsim 6.838s)
INFO    - 125/232 nsim_em7d_v22             tests/kernel/timer/timer_api/kernel.timer.tickless PASSED (nsim 8.776s)
INFO    - 126/232 nsim_em7d_v22             tests/kernel/spinlock/kernel.multiprocessing.spinlock SKIPPED (filter)
INFO    - 127/232 nsim_em7d_v22             tests/kernel/timer/timer_api/kernel.timer          PASSED (nsim 8.624s)
INFO    - 128/232 nsim_em7d_v22             tests/kernel/tickless/tickless_concept/kernel.tickless.concept PASSED (nsim 6.956s)
INFO    - 129/232 nsim_em7d_v22             tests/kernel/threads/thread_stack/kernel.threads.thread_stack PASSED (nsim 4.377s)
INFO    - 130/232 nsim_em7d_v22             tests/kernel/smp/kernel.multiprocessing.smp        SKIPPED (filter)
INFO    - 131/232 nsim_em7d_v22             tests/kernel/threads/no-multithreading/kernel.threads.no-multithreading PASSED (nsim 2.628s)
INFO    - 132/232 nsim_em7d_v22             tests/kernel/threads/thread_init/kernel.threads.init PASSED (nsim 3.271s)
INFO    - 133/232 nsim_em7d_v22             tests/kernel/threads/thread_apis/kernel.threads.apis PASSED (nsim 3.451s)
INFO    - 134/232 nsim_em7d_v22             tests/kernel/threads/dynamic_thread/kernel.threads.dynamic PASSED (nsim 2.548s)
INFO    - 135/232 nsim_em7d_v22             tests/kernel/sleep/kernel.common.timing            PASSED (nsim 2.653s)
INFO    - 136/232 nsim_em7d_v22             tests/kernel/semaphore/semaphore/kernel.semaphore  PASSED (nsim 2.877s)
INFO    - 137/232 nsim_em7d_v22             tests/kernel/sched/metairq/kernel.scheduler.metairq PASSED (nsim 2.476s)
INFO    - 138/232 nsim_em7d_v22             tests/kernel/sched/schedule_api/kernel.scheduler.multiq_no_timeslicing PASSED (nsim 3.081s)
INFO    - 139/232 nsim_em7d_v22             tests/kernel/sched/schedule_api/kernel.scheduler.no_timeslicing PASSED (nsim 3.145s)
INFO    - 140/232 nsim_em7d_v22             tests/kernel/sched/preempt/kernel.scheduler.preempt PASSED (nsim 3.896s)
INFO    - 141/232 nsim_em7d_v22             tests/kernel/sched/schedule_api/kernel.scheduler.multiq PASSED (nsim 17.524s)
INFO    - 142/232 nsim_em7d_v22             tests/kernel/sched/deadline/kernel.scheduler.deadline PASSED (nsim 2.277s)
INFO    - 143/232 nsim_em7d_v22             tests/kernel/sched/schedule_api/kernel.scheduler   PASSED (nsim 17.686s)
INFO    - 144/232 nsim_em7d_v22             tests/kernel/profiling/profiling_api/kernel.common.profiling PASSED (nsim 2.367s)
INFO    - 145/232 nsim_em7d_v22             tests/kernel/queue/kernel.queue                    PASSED (nsim 2.948s)
INFO    - 146/232 nsim_em7d_v22             tests/kernel/queue/kernel.queue.poll               PASSED (nsim 2.791s)
INFO    - 147/232 nsim_em7d_v22             tests/kernel/poll/kernel.poll                      PASSED (nsim 2.605s)
INFO    - 148/232 nsim_em7d_v22             tests/kernel/pending/kernel.objects                PASSED (nsim 3.158s)
INFO    - 149/232 nsim_em7d_v22             tests/kernel/pipe/pipe_api/kernel.pipe.api         PASSED (nsim 2.697s)
INFO    - 150/232 nsim_em7d_v22             tests/kernel/pipe/pipe/kernel.pipe                 PASSED (nsim 2.609s)
INFO    - 151/232 nsim_em7d_v22             tests/kernel/mp/kernel.multiprocessing             SKIPPED (filter)
INFO    - 152/232 nsim_em7d_v22             tests/kernel/obj_tracing/kernel.objects.tracing    PASSED (nsim 2.428s)
INFO    - 153/232 nsim_em7d_v22             tests/kernel/mutex/sys_mutex/system.mutex.nouser   PASSED (nsim 2.428s)
INFO    - 154/232 nsim_em7d_v22             tests/kernel/mem_protect/mem_map/kernel.memory_protection.mem_map SKIPPED (filter)
INFO    - 155/232 nsim_em7d_v22             tests/kernel/mutex/sys_mutex/system.mutex          PASSED (nsim 2.378s)
INFO    - 156/232 nsim_em7d_v22             tests/kernel/mutex/mutex_api/kernel.mutex          PASSED (nsim 2.464s)
INFO    - 157/232 nsim_em7d_v22             tests/kernel/msgq/msgq_api/kernel.message_queue    PASSED (nsim 2.633s)
INFO    - 158/232 nsim_em7d_v22             tests/kernel/mem_slab/mslab_threadsafe/kernel.memory_slabs.threadsafe PASSED (nsim 3.013s)
INFO    - 159/232 nsim_em7d_v22             tests/kernel/mem_slab/mslab_concept/kernel.memory_slabs.concept PASSED (nsim 2.348s)
INFO    - 160/232 nsim_em7d_v22             tests/kernel/mem_protect/userspace/kernel.memory_protection.userspace.gap_filling.arc SKIPPED (filter)
INFO    - 161/232 nsim_em7d_v22             tests/kernel/mem_slab/mslab_api/kernel.memory_slabs.api PASSED (nsim 2.360s)
INFO    - 162/232 nsim_em7d_v22             tests/kernel/mem_slab/mslab/kernel.memory_slabs.api PASSED (nsim 2.221s)
INFO    - 163/232 nsim_em7d_v22             tests/kernel/mem_protect/sys_sem/kernel.memory_protection.sys_sem.nouser PASSED (nsim 2.714s)
INFO    - 164/232 nsim_em7d_v22             tests/kernel/mem_protect/sys_sem/kernel.memory_protection.sys_sem PASSED (nsim 2.552s)
INFO    - 165/232 nsim_em7d_v22             tests/kernel/mem_protect/futex/kernel.futex        PASSED (nsim 2.593s)
INFO    - 166/232 nsim_em7d_v22             tests/kernel/mem_protect/userspace/kernel.memory_protection.userspace PASSED (nsim 2.999s)
INFO    - 167/232 nsim_em7d_v22             tests/kernel/mem_protect/mem_protect/kernel.memory_protection.gap_filling SKIPPED (filter)
INFO    - 168/232 nsim_em7d_v22             tests/kernel/mem_protect/syscalls/kernel.memory_protection.syscalls PASSED (nsim 19.524s)
INFO    - 169/232 nsim_em7d_v22             tests/kernel/mem_protect/stack_random/kernel.memory_protection.stack_random PASSED (nsim 2.478s)
INFO    - 170/232 nsim_em7d_v22             tests/kernel/mem_protect/protection/kernel.memory_protection.protection PASSED (nsim 2.466s)
INFO    - 171/232 nsim_em7d_v22             tests/kernel/mem_protect/stackprot/kernel.memory_protection.stackprot PASSED (nsim 2.597s)
INFO    - 172/232 nsim_em7d_v22             tests/kernel/mem_protect/obj_validation/kernel.memory_protection.obj_validation PASSED (nsim 2.439s)
INFO    - 173/232 nsim_em7d_v22             tests/kernel/mem_pool/mem_pool_threadsafe/kernel.memory_pool.threadsafe.legacy PASSED (nsim 2.317s)
INFO    - 174/232 nsim_em7d_v22             tests/kernel/mem_protect/mem_protect/kernel.memory_protection PASSED (nsim 3.248s)
INFO    - 175/232 nsim_em7d_v22             tests/kernel/mem_pool/sys_mem_pool/kernel.memory_pool.sys PASSED (nsim 3.817s)
INFO    - 176/232 nsim_em7d_v22             tests/kernel/mem_pool/mem_pool_threadsafe/kernel.memory_pool.threadsafe PASSED (nsim 2.364s)
INFO    - 177/232 nsim_em7d_v22             tests/kernel/mem_pool/mem_pool_concept/kernel.memory_pool.concept.legacy PASSED (nsim 2.429s)
INFO    - 178/232 nsim_em7d_v22             tests/kernel/mem_pool/mem_pool_concept/kernel.memory_pool.concept PASSED (nsim 2.309s)
INFO    - 179/232 nsim_em7d_v22             tests/kernel/mem_pool/mem_pool_api/kernel.memory_pool.api.legacy PASSED (nsim 2.372s)
INFO    - 180/232 nsim_em7d_v22             tests/kernel/mem_pool/mem_pool_api/kernel.memory_pool.api PASSED (nsim 2.403s)
INFO    - 181/232 nsim_em7d_v22             tests/kernel/mem_pool/mem_pool/kernel.memory_pool.legacy PASSED (nsim 2.359s)
INFO    - 182/232 nsim_em7d_v22             tests/kernel/mem_pool/mem_pool/kernel.memory_pool  PASSED (nsim 3.285s)
INFO    - 183/232 nsim_em7d_v22             tests/kernel/mem_heap/mheap_api_concept/kernel.memory_heap PASSED (nsim 2.323s)
INFO    - 184/232 nsim_em7d_v22             tests/kernel/gen_isr_table/arch.interrupt.gen_isr_table.arc SKIPPED (filter)
INFO    - 185/232 nsim_em7d_v22             tests/kernel/gen_isr_table/arch.interrupt.gen_isr_table.arm_mainline SKIPPED (filter)
INFO    - 186/232 nsim_em7d_v22             tests/kernel/mbox/mbox_usage/kernel.mailbox.usage  PASSED (nsim 2.394s)
INFO    - 187/232 nsim_em7d_v22             tests/kernel/mbox/mbox_api/kernel.mailbox.api      PASSED (nsim 2.406s)
INFO    - 188/232 nsim_em7d_v22             tests/kernel/lifo/lifo_usage/kernel.lifo.usage     PASSED (nsim 2.448s)
INFO    - 189/232 nsim_em7d_v22             tests/kernel/lifo/lifo_api/kernel.lifo             PASSED (nsim 2.473s)
INFO    - 190/232 nsim_em7d_v22             tests/kernel/fifo/fifo_usage/kernel.fifo.usage.poll PASSED (nsim 2.216s)
INFO    - 191/232 nsim_em7d_v22             tests/kernel/interrupt/arch.interrupt              PASSED (nsim 4.932s)
INFO    - 192/232 nsim_em7d_v22             tests/kernel/fifo/fifo_usage/kernel.fifo.usage     PASSED (nsim 2.330s)
INFO    - 193/232 nsim_em7d_v22             tests/kernel/fifo/fifo_timeout/kernel.fifo.timeout.poll PASSED (nsim 2.252s)
INFO    - 194/232 nsim_em7d_v22             tests/kernel/fifo/fifo_timeout/kernel.fifo.timeout PASSED (nsim 2.248s)
INFO    - 195/232 nsim_em7d_v22             tests/kernel/fifo/fifo_api/kernel.fifo.poll        PASSED (nsim 2.526s)
INFO    - 196/232 nsim_em7d_v22             tests/kernel/fifo/fifo_api/kernel.fifo             PASSED (nsim 2.441s)
INFO    - 197/232 nsim_em7d_v22             tests/kernel/early_sleep/kernel.common.sleep       PASSED (nsim 2.420s)
INFO    - 198/232 nsim_em7d_v22             tests/kernel/context/kernel.common                 PASSED (nsim 2.620s)
INFO    - 199/232 nsim_em7d_v22             tests/kernel/device/kernel.device.pm               PASSED (nsim 2.403s)
INFO    - 200/232 nsim_em7d_v22             tests/kernel/device/kernel.device                  PASSED (nsim 2.465s)
INFO    - 201/232 nsim_em7d_v22             tests/kernel/common/kernel.common.misra            PASSED (nsim 3.425s)
INFO    - 202/232 nsim_em7d_v22             tests/drivers/uart/uart_async_api/drivers.uart.uart_async_api.rtt SKIPPED (filter)
INFO    - 203/232 nsim_em7d_v22             tests/drivers/sensor/generic/driver.sensor         PASSED (nsim 2.498s)
INFO    - 204/232 nsim_em7d_v22             tests/drivers/uart/uart_async_api/drivers.uart.uart_async_api SKIPPED (filter)
INFO    - 205/232 nsim_em7d_v22             tests/kernel/common/kernel.common                  PASSED (nsim 3.341s)
INFO    - 206/232 nsim_em7d_v22             tests/drivers/uart/uart_basic_api/drivers.uart.poll PASSED (build)
INFO    - 207/232 nsim_em7d_v22             tests/drivers/console/drivers.console.uart         PASSED (nsim 2.327s)
INFO    - 208/232 nsim_em7d_v22             tests/drivers/uart/uart_basic_api/drivers.uart     PASSED (build)
INFO    - 209/232 nsim_em7d_v22             tests/drivers/uart/uart_basic_api/drivers.uart.shell PASSED (build)
INFO    - 210/232 nsim_em7d_v22             tests/drivers/entropy/api/drivers.entropy          SKIPPED (filter)
INFO    - 211/232 nsim_em7d_v22             tests/crypto/rand32/crypto.rand32.random_ctr_drbg  SKIPPED (filter)
INFO    - 212/232 nsim_em7d_v22             tests/drivers/ipm/drivers.ipc.mailbox              PASSED (nsim 2.406s)
INFO    - 213/232 nsim_em7d_v22             tests/crypto/rand32/crypto.rand32.random_hw_xoroshiro SKIPPED (filter)
INFO    - 214/232 nsim_em7d_v22             tests/drivers/hwinfo/api/drivers.device_id         PASSED (nsim 2.257s)
INFO    - 215/232 nsim_em7d_v22             tests/crypto/tinycrypt_hmac_prng/crypto.tinycrypt.hmac_prng PASSED (nsim 11.529s)
INFO    - 216/232 nsim_em7d_v22             tests/crypto/rand32/crypto.rand32                  PASSED (nsim 2.371s)
INFO    - 217/232 nsim_em7d_v22             tests/crypto/rand32/crypto.rand32.random_sw_systimer PASSED (nsim 2.447s)
INFO    - 218/232 nsim_em7d_v22             tests/benchmarks/footprints/benchmark.kernel.footprints.userspace PASSED (build)
INFO    - 219/232 nsim_em7d_v22             tests/benchmarks/footprints/benchmark.kernel.footprints.default PASSED (build)
INFO    - 220/232 nsim_em7d_v22             tests/benchmarks/cmsis_dsp/basicmath/benchmark.cmsis_dsp.basicmath SKIPPED (filter)
INFO    - 221/232 nsim_em7d_v22             tests/benchmarks/mbedtls/benchmark.crypto.mbedtls  SKIPPED (filter)
INFO    - 222/232 nsim_em7d_v22             tests/benchmarks/latency_measure/benchmark.kernel.latency.stm32 SKIPPED (filter)
INFO    - 223/232 nsim_em7d_v22             tests/benchmarks/data_structure_perf/rbtree_perf/benchmark.data_structures PASSED (nsim 2.365s)
INFO    - 224/232 nsim_em7d_v22             tests/benchmarks/data_structure_perf/dlist_perf/benchmark.data_structures PASSED (nsim 2.487s)
INFO    - 225/232 nsim_em7d_v22             tests/benchmarks/sys_kernel/benchmark.kernel.core  PASSED (nsim 2.378s)
INFO    - 226/232 nsim_em7d_v22             tests/benchmarks/sched/benchmark.kernel.scheduler  PASSED (build)
INFO    - 227/232 nsim_em7d_v22             tests/crypto/tinycrypt/crypto.tinycrypt            FAILED Timeout (nsim 300.515s)
ERROR   - see: /SCRATCH/watson/zephyr_project/zephyr/build.sanity.nsim_em7d_v22/nsim_em7d_v22/tests/crypto/tinycrypt/crypto.tinycrypt/handler.log
INFO    - 228/232 nsim_em7d_v22             tests/crypto/mbedtls/crypto.mbedtls                PASSED (nsim 87.625s)
INFO    - 229/232 nsim_em7d_v22             tests/application_development/gen_inc_file/buildsystem.include_file PASSED (nsim 2.378s)
INFO    - 230/232 nsim_em7d_v22             tests/application_development/libcxx/application_development.cpp.libcxx PASSED (nsim 2.289s)
INFO    - 231/232 nsim_em7d_v22             tests/application_development/libcxx/application_development.cpp.libcxx.exceptions PASSED (nsim 2.278s)
INFO    - 232/232 nsim_em7d_v22             tests/application_development/cpp/application_development.cpp.main PASSED (nsim 2.185s)

INFO    - 181 of 182 tests passed (99.45%), 1 failed, 746 skipped with 0 warnings in 2301.70 seconds
INFO    - In total 1176 test cases were executed on 1 out of total 291 platforms (0.34%)
INFO    - 163 tests executed on platforms, 19 tests were only built.
```

ruuddw reviewed (approved):
Thanks, adding this configuration must have gone parallel to the switch to dwuart...

abrodkin reviewed (changes requested):
Please explain in the commit message what was wrong with existing configuration, mention @andrewboie's report and explain why switching to a standard UART fixes reported problem:
1. We moved all nSIM boards to new UART via Kconfig option (add a reference to the commit which implemented it) and (not sure that requires an additional check)
2. Dropped ARC UART from Zephyr at all (if that was done, also add a commit as reference here).

IRISZZW committed:
boards: arc: nsim_em7d_v22: switch to ns16550 UART model

In PR #26836, we switch nSIM from custom legacy ARC UART model
to ns16550 model, which will allow us to use zephyr images build for
nSIM on other platforms like HAPS, QEMU, etc...
In PR #27334, which introduce new board nsim_em7d_v22, has gone
parallel to the switch to dwuart, and is still using legacy model.
With wrong configuration, the uart for nsim_em7d_v22 has no output,
which cause all tests failure.

Signed-off-by: Watson Zeng <zhiwei@synopsys.com>

IRISZZW commented:
> Please explain in the commit message what was wrong with existing configuration, mention @andrewboie's report and explain why switching to a standard UART fixes reported problem:
> 
> 1. We moved all nSIM boards to new UART via Kconfig option (add a reference to the commit which implemented it) and (not sure that requires an additional check)
> 2. Dropped ARC UART from Zephyr at all (if that was done, also add a commit as reference here).

Have dropped ARC UART from Zephyr at all.

abrodkin reviewed (approved):
I'd add an empty line between 2 sentences for better readability. Otherwise LGTM, thanks!

tyrantlizard77 reviewed (approved):
verified, thanks
